sstables/sortedmap_key_index.go

```diff
@@ -0,0 +1,173 @@
+package sstables
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	rProto "github.com/thomasjungblut/go-sstables/recordio/proto"
+	"github.com/thomasjungblut/go-sstables/skiplist"
+	"github.com/thomasjungblut/go-sstables/sstables/proto"
+)
+
+// SortedMapIndex is keeping the entire index as a map and the ordered keys as a slice in memory
+// Use binary search to iter over the index.
+type ByteKeyMapper[T comparable] interface {
+	MapBytes(data []byte) T
+}
+
+// use this key Mapper for sha1 keys
+type Byte20KeyMapper[T comparable] struct{}
+
+func (m Byte20KeyMapper[T]) MapBytes(data []byte) T {
+	var result [20]byte
+
+	// Gestion de la longueur
+	if len(data) > 20 {
+		data = data[:20]
+	}
+
+	// Copie des données
+	copy(result[:], data)
+
+	// Conversion générique - nécessite que T soit [20]byte
+	return any(result).(T)
+}
```

Comment:
anglais, s'il vous plaît :) 

looks good, I would expect the type to be just `Byte20KeyMapper` and the func to be `MapBytes(data []byte) `[20]byte`, that should be enough for the compiler to match the signatures.

I'd maybe also panic if there are more than twenty bytes in there.

Reply:
oups, sorry for the french. I tried without the T generic, but it doesn't work.

Reply:
no worries at all! I'll try today, I just need to finish up some stuff

Reply:
```
type ByteKeyMapper[T comparable] interface {
	MapBytes(data []byte) T
}

// use this key Mapper for sha1 keys
type Byte20KeyMapper struct{}

func (m Byte20KeyMapper) MapBytes(data []byte) [20]byte {
	var result [20]byte

	if len(data) > 20 {
		data = data[:20]
	}
	copy(result[:], data)
	return result
}
```

seems to compile for me